bindings/container.cpp

```diff
@@ -201,6 +201,14 @@ App* Container::loadAndInstantiate(QString path) {
         std::cout << QString("%1 is not a valid Holochain DNA file!").arg(path).toStdString() << std::endl;
         return 0;
     } else {
-        return new App(dna, QStandardPaths::writableLocation(QStandardPaths::TempLocation), this);
+        QString dna_json = QString(holochain_dna_to_json(dna));
+        QCryptographicHash hash(QCryptographicHash::Sha256);
+        hash.addData(dna_json.toUtf8());
```

Comment:
What does this line do?

Reply:
This seems to be a hash factory. So you can add data to as much as you like and then call `result()` to actually generate the hash

Reply:
exactly